bottlerocket/agents/src/userdata.rs

```diff
@@ -0,0 +1,158 @@
+// This is only used in the build script for now, but it's in a separate crate so we can have unit
+// tests.
```

Comment:
```suggestion
// This is only used in the build script for now, but it's in a separate crate so we can have unit
// tests.
```
I don't think this comment applies here.

After looking through this, I think we should instead switch to using the `merge-toml` library from the bottlerocket repo directly https://github.com/bottlerocket-os/bottlerocket/blob/67e5abfb0e1575fb6bcb7a710cdb3d422ef12283/sources/api/storewolf/merge-toml/src/lib.rs like @ecpullen mentioned before. We can still use the custom error types and keep `decode_to_string` somewhere else/

Reply:
I can remove the comment, but copying the code instead of using the library is something that was discussed and @webern especially felt like we should not rely on a library from a different project that was never meant to be included in other projects.

Reply:
This is true. I am hesitant to create the first ever external dependency on library code that resides in Bottlerocket. I'd rather we take an action item to create a merge_toml crate external to both projects and put it on crates.io than to have dependencies going in both directions (Bottlerocket depends on testsys which depends on Bottlerocket).

Reply:
Can you add a comment to add some attribution for this code? Just to keep track that it's copied from bottlerocket. Adding the permalink and saying this is directly borrow from there is probably enough. https://github.com/bottlerocket-os/bottlerocket/blob/develop/sources/api/storewolf/merge-toml/src/lib.rs

Reply:
Can you also create an issue in both repos so we make sure this gets fixed.